packages/notification-service/src/invites/invites.ts

```diff
@@ -7,27 +8,36 @@ import { getContractKit } from '../util/utils'
 
 const TAG = 'INVITES'
 
-export async function handleInvites() {
-  const kit = await getContractKit()
-  const fromBlock = getLastInviteBlockNotified() + 1
-  if (fromBlock <= 0) {
+function updateLastInviteBlock(fromBlock: number, events: EventLog[]) {
+  if (events.length === 0) {
     return
   }
-  console.debug(TAG, `Starting to fetch invites from block ${fromBlock}`)
+  const maxBlock = events.reduce((max, event) => Math.max(max, event.blockNumber), fromBlock)
+  setLastInviteBlockNotified(maxBlock)
+}
 
-  const escrow = await kit.contracts.getEscrow()
-  const events = await escrow.getPastEvents('Withdrawal', {
-    fromBlock,
-  })
-  console.debug(TAG, `Got ${events.length} escrow withdrawal events`)
-  let maxBlock = fromBlock
-  for (const event of events) {
-    maxBlock = Math.max(event.blockNumber, fromBlock)
-    const inviter = event.returnValues.to.toLowerCase()
-    console.debug(TAG, `Sending invite notification to ${inviter}`)
-    await sendInviteNotification(inviter)
-  }
-  if (maxBlock > fromBlock) {
-    setLastInviteBlockNotified(maxBlock)
+export async function handleInvites() {
+  try {
+    const kit = await getContractKit()
+    const fromBlock = getLastInviteBlockNotified() + 1
+    if (fromBlock <= 0) {
+      return
+    }
+    console.debug(TAG, `Starting to fetch invites from block ${fromBlock}`)
+
+    const escrow = await kit.contracts.getEscrow()
+    const events = await escrow.getPastEvents('Withdrawal', {
```

Comment:
One more question: do we have any guarantees regarding the order of events returned? I assume not but just double checking

Reply:
Yeah, that's why I prefer using Forno than Blockscout. Blockscout has no such guarantees but Forno does return in order. If it gets backed up it will get backed up in order at least.